### PR TITLE
optimise WikiDocument indexing

### PR DESCRIPTION
**Pull request by LeoMcA**
https://github.com/mozilla/sumo-project/issues/695

### CODE REVIEW

File: kitsune/search/v2/base.py [hunk 1]
```diff
@@ -74,15 +67,13 @@ def prepare(cls, instance, **kwargs):
                     # https://docs.djangoproject.com/en/2.2/ref/utils/#django.utils.timezone.make_aware
                     value = timezone.make_aware(value, is_dst=False).astimezone(timezone.utc)
 
-                if use_parent_obj:
-                    value = obj.get_field_value(f, instance.parent, prepare_method)
-
-                setattr(obj, f, value)
+                if not parent_id:
```
Comment (akatsoulas) [inline kitsune/search/v2/base.py:70]: when we want to merge the docs, we are not ignoring the value of the object but rather we are overwriting with the one of the parent where applicable. The removed snippet is using attrs from both parent and child but the new one does not

**Review by akatsoulas — CHANGES_REQUESTED** [review]

File: kitsune/search/v2/base.py (hunk 1 above)
Comment (LeoMcA) [reply kitsune/search/v2/base.py:70]: I'm not sure I understand.

If we have parent P, child C, and non-locale-field foobar, previously:

```
WikiDocument.prepare(P).foobar == P.foobar
WikiDocument.prepare(C).foobar == P.foobar
```

now:

```
WikiDocument.preapre(P).foobar == P.foobar
WikiDocument.prepare(C).foobar raises AttributeError
```

Comment (akatsoulas) [reply kitsune/search/v2/base.py:70]: exactly :+1: 

The only data that we want to keep from the child are the locale aware fields, Everything else should default to the attrs of the parent.

Comment (akatsoulas) [reply kitsune/search/v2/base.py:70]: This was messing with fields like `locale` which was pointing to the one of the child when it should have pointed to the one of the parent. The fields though that was causing problems were either removed or updated in #4574 so we could actually proceed with this and avoid preparing field values twice.

**Review by akatsoulas — APPROVED** [review]

**Pull request merged**